resource/src/lib.rs

```diff
@@ -195,7 +192,11 @@ fn path_as_key(path: &PathBuf) -> Cow<str> {
 }
 
 fn file_system(path: PathBuf) -> Option<Resource> {
-    path.canonicalize().ok().map(Resource::FileSystem)
+    if path.exists() {
+        Some(Resource::FileSystem(path))
+    } else {
+        None
+    }
```

Comment:
I think it will (will let @doitian confirm), but either way, this PR is about preserving original behavior as much as we can while fixing the windows build, from this point of view, I'd say we are all good since the change here doesn't alter original behaviors

Reply:
The original only attempt to canonicalize the path, but not check the existence. So I think it is different.

Reply:
Rust's canonicalize has a side effect: when the passed path does not exist, it will instead return error, which is then converted to None here

Reply:
Oh, I see...